appvm-scripts/usrbin/qubes-run-xorg

```diff
@@ -94,14 +96,31 @@ chmod 770 /var/run/xf86-qubes-socket
 # 2**30
 echo 1073741824 > /sys/module/xen_gntalloc/parameters/limit
 
-if [ -x /bin/loginctl ]; then
-    # logind needs at least one device with "master-of-seat" tag to
-    # consider this seat as graphical; we use pcspkr, for which we ship udev
-    # rules adding that tag
-    loginctl attach "$XDG_SEAT" /sys/devices/platform/pcspkr/input/*
-fi
+if qsvc guivm-gui-agent; then
+    DISPLAY_XORG=:1
+    DISPLAY_XEPHYR=:0
+
+    # Create Xorg and Xephyr server (emulated GPU passthrough)
+    runuser -u "$DEFAULT_USER" -- /bin/sh -l -c "exec /usr/bin/xinit /usr/bin/qubes-run-xephyr -- $XORG $DISPLAY_XORG -nolisten tcp vt07 -wr -config xorg-qubes.conf > ~/.xorg-errors 2>&1" &
+    # Wait Xephyr display available
+    while ! [ -e /tmp/.X11-unix/X0 ]
+    do
+        sleep 0.1
+    done
+    # Run xsession
+    /usr/bin/qubes-gui-runuser "$DEFAULT_USER" /bin/sh -l -c "DISPLAY=$DISPLAY_XEPHYR exec $XSESSION > ~/.xsession-errors 2>&1" &
```

Comment:
Any specific reason for sending it into background? Maybe even `exec` as the other case would work here?